GCC 15 Build issues with ESMF 8.8.1

ESMF Team,

Today I tried building ESMF 8.8.1 using GCC 15.1 on a Linux box and it failed with:
```
mpicxx -c -std=c++11 -O -DNDEBUG -fPIC -m64 -mcmodel=small -pthread  -fopenmp -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/../include -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/build_config/Linux.gfortran.default -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Infrastructure -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Superstructure -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Infrastructure/stubs/pthread  -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/include  -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/x86_64-pc-linux-gnu/gfortran_15.1.0-openmpi_5.0.4/Linux/include/netcdf -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/include -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_VERSION_STRING_GIT='"v8.8.1"' -DESMF_LOWERCASE_SINGLEUNDERSCORE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_LAPACK_INTERNAL=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_NETCDF=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_PIO=1 -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTPERFORMANCE -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=openmpi -DESMF_DIR=/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf -D__SDIR__='"src/prologue/yaml-cpp/src"' -DESMF_CXXSTD=11 -DESMF_NO_SIGUSR2 /ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/scantag.cpp -o /ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/obj/objO/Linux.gfortran.64.openmpi.default/scantag.o
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:221:11: error: ‘uint16_t’ was not declared in this scope
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |           ^~~~~~~~
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:13:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   12 | #include "yaml-cpp/null.h"
  +++ |+#include <cstdint>
   13 | #include "yaml-cpp/ostream_wrapper.h"
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:221:21: error: ‘uint16_t’ was not declared in this scope
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |                     ^~~~~~~~
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:221:21: note: ‘uint16_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:221:29: error: template argument 1 is invalid
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |                             ^
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:221:29: error: template argument 2 is invalid
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp: In function ‘int YAML::Utils::{anonymous}::EncodeUTF16SurrogatePair(int)’:
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:222:9: error: ‘uint32_t’ does not name a type
  222 |   const uint32_t leadOffset = 0xD800 - (0x10000 >> 10);
      |         ^~~~~~~~
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:222:9: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:225:5: error: ‘leadOffset’ was not declared in this scope
  225 |     leadOffset | (codePoint >> 10),
      |     ^~~~~~~~~~
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:227:3: error: cannot convert ‘<brace-enclosed initializer list>’ to ‘int’ in return
  227 |   };
      |   ^
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp: In function ‘void YAML::Utils::{anonymous}::WriteDoubleQuoteEscapeSequence(YAML::ostream_wrapper&, int, YAML::StringEscaping::value)’:
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:246:55: error: request for member ‘first’ in ‘surrogatePair’, which is of non-class type ‘int’
  246 |     WriteDoubleQuoteEscapeSequence(out, surrogatePair.first, stringEscapingStyle);
      |                                                       ^~~~~
/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/emitterutils.cpp:247:55: error: request for member ‘second’ in ‘surrogatePair’, which is of non-class type ‘int’
  247 |     WriteDoubleQuoteEscapeSequence(out, surrogatePair.second, stringEscapingStyle);
      |                                                       ^~~~~~
mpicxx -c -std=c++11 -O -DNDEBUG -fPIC -m64 -mcmodel=small -pthread  -fopenmp -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/../include -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/build_config/Linux.gfortran.default -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Infrastructure -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Superstructure -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/Infrastructure/stubs/pthread  -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/include  -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/x86_64-pc-linux-gnu/gfortran_15.1.0-openmpi_5.0.4/Linux/include/netcdf -I/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/include -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_VERSION_STRING_GIT='"v8.8.1"' -DESMF_LOWERCASE_SINGLEUNDERSCORE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_LAPACK_INTERNAL=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_NETCDF=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_PIO=1 -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTPERFORMANCE -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=openmpi -DESMF_DIR=/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf -D__SDIR__='"src/prologue/yaml-cpp/src"' -DESMF_CXXSTD=11 -DESMF_NO_SIGUSR2 /ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/src/prologue/yaml-cpp/src/directives.cpp -o /ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/obj/objO/Linux.gfortran.64.openmpi.default/directives.o
make[11]: *** [/ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/build/common.mk:4047: /ford1/share/gmao_SIteam/Baselibs/ESMA-Baselibs-8.15.0-TEST/src/esmf/obj/objO/Linux.gfortran.64.openmpi.default/emitterutils.o] Error 1
```

If I had to guess, g++ 15.1.0 is now a bit stricter in some sense.

But of course, this is in yaml-cpp it seems which I guess is "external" to ESMF. I'd gladly try any changes you'd like me to. I do *not* know enough C++ to try and fix this.

--- a/src/prologue/yaml-cpp/src/emitterutils.cpp
+++ b/src/prologue/yaml-cpp/src/emitterutils.cpp
@@ -1,4 +1,5 @@
 #include <algorithm>
+#include <cstdint>
 #include <iomanip>
 #include <sstream>
 

